schemas/edm.xsd

```diff
@@ -648,16 +642,15 @@
       <!-- The below pattern represents the allowed identifiers in the ECMAScript specification plus the '/' for path segment separation
         and
         the '.' for namespace qualification inside the segments. -->
-      <xs:pattern
-        value="(/?@?[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}]{0,}(([\./#@]|/@)[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}]{0,}){0,}(/$count)?)?" />
+      <xs:pattern value="(/?@?[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}]{0,}(([\./#@]|/@)[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}]{0,}){0,}(/$count)?)?" />
     </xs:restriction>
   </xs:simpleType>
   <xs:simpleType name="TTarget">
     <xs:restriction base="xs:string">
       <!-- The below pattern represents the allowed identifiers in the ECMAScript specification plus the '.' for namespace qualification,
         the '/' for path segment separation, and parentheses and comma for overload targeting -->
       <xs:pattern
-        value="[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}]{0,}(([\.\(,#]|/@?)(\)|[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}]{0,}\)?)){0,}(/$ReturnType)?" />
+        value="[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}]{0,}(([.,#(]|/@?|\(?\){1,}((,|/@?))?)[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}]{0,}){0,}\(?\){0,}(/$ReturnType)?" />

```

Comment:
The edm.xsd should _clarify_ the CSDL syntax. With patterns like these it rather _obscures_ the syntax, or couches it in riddles. Perhaps this is another justification for pre-processing attributes into child nodes: Convert
```
<Annotations Target="One.OddWaldos(Collection(One.Waldo),One.Waldo)">
  ...
</Annotations>
```
into
```
<Annotations>
  <FunctionTarget Name="One.OddWaldos">
    <ParameterType Name="One.Waldo" Collection="true"/>
    <ParameterType Name="One.Waldo"/>
  </FunctionTarget>
  ...
</Annotations>
```
and have this XML validated by the schema.

Reply:
We do not want to build an EDMX validator, we want to provide correct XSD files for EDMX.